src/odemis/util/polar.py

```diff
@@ -66,81 +56,138 @@ def AngleResolved2Polar(data, output_size, hole=True, dtype=None):
-    # FIXME: need rotation (=swap axes), but swapping theta/phi slows down the
-    # interpolation by 3 ?!
-    with warnings.catch_warnings():
-        # Some points might be so close that they are identical (within float
-        # precision). It's fine, no need to generate a warning.
-        warnings.simplefilter("ignore", DuplicatePointWarning)
-        triang = Triangulation(theta_data.flat, phi_data.flat)  # FIXME: Leaks memory when run in a separate thread
-    interp = triang.linear_interpolator(omega_data.flat, default_value=0)
-    qz = interp[-h_output_size:h_output_size:complex(0, output_size),  # Y
-                - h_output_size:h_output_size:complex(0, output_size)]  # X
-    qz = qz.swapaxes(0, 1)[:, ::-1]  # rotate by 90°
-
-# TODO use the standard matplotlib.tri.Triangulation, but it uses 3x more memory
-# also leaks, and is slower!
-#     triang = Triangulation(phi_data.flat, theta_data.flat)
-#     interp = LinearTriInterpolator(triang, omega_data.flat)
-#     xi, yi = numpy.meshgrid(numpy.linspace(-h_output_size, h_output_size, output_size),
-#                             numpy.linspace(-h_output_size, h_output_size, output_size))
-#     qz = interp(xi, yi)
-#     qz = qz[:, ::-1]
-
-# TODO: try scipy.spatial.Delaunay
+    theta = theta_data_masked * (h_output_size / math.pi * 2)
+    phi = phi_data_masked
+    theta_data_polar = numpy.cos(phi) * theta
+    phi_data_polar = numpy.sin(phi) * theta
+
+    # Multiple theta-phi combinations will be mapped to the same px in the output image after polar-transformation.
+    # Therefore, not all px in the output image are populated.
+    # Moreover, the data is masked with the mirror shape (mask_circle).
+    # Therefore, we perform a delaunay triangulation of the given data points.
+    # The delaunay object is passed to the interpolator with the corresponding intensity values (omega_data).
+    # The interpolator also receives a meshgrid (set of coordinates) of the size specified for the output image.
+    # The input data points (theta and phi) are mapped on the meshgrid. As the meshgrid contains much more positions
+    # compared to the input data points, the interpolator fills up the empty grid positions with intensity values.
+    # These intensity values are interpolated from the intensity values of the positions spanning the triangle they
+    # are contained in (triangle from delaunay triangulation).
+    # Grid positions located outside of any delaunay triangle are set to NaN.
+    triang = DelaunayTriangulation(numpy.array([theta_data_polar.flat[:], phi_data_polar.flat[:]]).T)
+    # create interpolation object
+    interp = LinearNDInterpolator(triang, omega_data_masked.flat)
+    # create grid of positions for interpolation
+    xi, yi = numpy.meshgrid(numpy.linspace(-h_output_size, h_output_size, output_size),
+                            numpy.linspace(-h_output_size, h_output_size, output_size))
+    # interpolate
+    _qz = interp(xi, yi)
```

Comment:
If you don't need `_qz` after, just call it `qz`, and one advantage is that it will make really sure that this array is discarded from memory as soon as it's not needed anymore.